Model/Ui/AdyenGenericConfigProvider.php

```diff
@@ -23,6 +23,32 @@ class AdyenGenericConfigProvider implements ConfigProviderInterface
 {
     const CODE = 'adyen_abstract';
 
+    // Seperate payment method codes
+    const ADYEN_IDEAL_CODE = 'adyen_ideal';
+    const ADYEN_AMAZONPAY_CODE = 'adyen_amazonpay';
+    const ADYEN_APPLEPAY_CODE = 'adyen_applepay';
+    const ADYEN_BCMC_MOBILE_CODE = 'adyen_bcmc_mobile';
+    const ADYEN_DOTPAY_CODE = 'adyen_dotpay';
+    const ADYEN_FACILYPAY_3X_CODE = 'adyen_facilypay_3x';
+    const ADYEN_MULTIBANCO_CODE = 'adyen_multibanco';
+    const ADYEN_GOOGLEPAY_CODE = 'adyen_googlepay';
+    const ADYEN_KLARNA_CODE = 'adyen_klarna';
+    const ADYEN_PAYPAL_CODE = 'adyen_paypal';
+    const ADYEN_SEPADIRECTDEBIT_CODE = 'adyen_sepadirectdebit';
+
+    // Separate payment method tx_variants
+    const IDEAL_TX_VARIANT = 'ideal';
+    const AMAZONPAY_TX_VARIANT = 'amazonpay';
+    const APPLEPAY_TX_VARIANT = 'applepay';
+    const BCMC_MOBILE_TX_VARIANT = 'bcmc_mobile';
+    const DOTPAY_TX_VARIANT = 'dotpay';
+    const FACILYPAY_3X_TX_VARIANT = 'facilypay_3x';
+    const MULTIBANCO_TX_VARIANT = 'multibanco';
+    const GOOGLEPAY_TX_VARIANT = 'googlepay';
+    const KLARNA_TX_VARIANT = 'klarna';
+    const PAYPAL_TX_VARIANT = 'paypal';
+    const SEPADIRECTDEBIT_TX_VARIANT = 'sepadirectdebit';
+
```

Comment:
can we got those constant from Adyen php lib?

Reply:
Hello @hossam-adyen, thank you for your review. Currently, it is not possible to obtain this information from PHP API Library. `adyen_XXXX` notation is specific to Magento. Obtaining tx_variants from `paymentMethods` call is possible, but they need to be statically defined in `di.xml` file.